org.eclipse.jdt.launching/launching/org/eclipse/jdt/internal/launching/ClasspathShortener.java

```diff
@@ -302,6 +302,7 @@ private void shortenPathUsingClasspathArgumentFile(int modulePathArgumentIndex,
 						+ systemCharset.displayName() + ".", null)); //$NON-NLS-1$
 			}
 			Files.writeString(argFile.toPath(), arg, systemCharset);
+			argFile.deleteOnExit();
```

Comment:
While this is an improvement it would not solve the issue that developers complain that the file shows up in the git staging view while they have eclipse open. So they could still accidentally commit it.

Reply:
That's true, and this is why this is still a draft ...